internal/provider/resource_basistheory_application_key.go

```diff
@@ -127,7 +128,7 @@ func resourceApplicationKeyDelete(ctx context.Context, data *schema.ResourceData
 	keyId := data.Id()
 	response, err := basisTheoryClient.ApplicationKeysApi.Delete(ctxWithApiKey, applicationId, keyId).Execute()
 
-	if err != nil {
+	if err != nil && !strings.Contains(err.Error(), "Not Found") {
```

Comment:
There were some flakey test failures regarding this. I presume there was a race condition between deleting the Application (and cascading key deletes) and also explicitly deleting ApplicationKeys.